### DISCUSSION TITLE
What is the best way to deal with numbers wrapped in Decimal.js objects?

### DISCUSSION THREAD

**Discussion by tomashaddad** [prisma/prisma#16218] [Q&A] [answered]
This relates to #12471 and #6049.

I understand the rationale behind wrapping numbers in Decimal.js objects (`number` not being enough to deal with arbitrary precision types), but their default treatment as a `string` is annoying. I ran into the same problem as described in those issues, except with Postgres and `numeric` types (an alias for `decimal`). I incorrectly assumed they would be of type `number` in my front end, and attempting to use them obviously failed without a `Number()` cast.

What is the best way to deal with this? The second issue includes a solution overriding the `toJSON` function of `Decimal` to return a number rather than a string, but this is obviously not ideal. My current solution is to just iterate through the response and casting manually before returning it to the front end, but I would love to just tell Prisma to give me a `number` rather than a `Decimal`.

**Comment by nurul3101** [comment, marked as answer]
Hello @tomashaddad 👋 

I have seen that it is common to return decimals from the database as strings as to not lose precision, as returning as number would lose precision/decimal places. There were certain issues reported like #10412 , [#9163](https://github.com/prisma/prisma/issues/9163) where precision was lost due to the behaviour of sending number instead of string.

In case you require a number, the solution would be to do something like this:
```
const prisma = require('@prisma/client');

prisma.Decimal.prototype.toJSON = function() {
  return this.toNumber();
}
```

**Reply by vladi-stevanovic** [reply]
👋 Hi @tomashaddad , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
